# Descriptive Statistics - Homework 1

Completed by **Martin Breuss**, Bsc.

[ma1531br-s@student.lu.se](mailto:ma1531br-s@student.lu.se)

Code available at 

In [1]:
import numpy as np

## Question 1

The rainfall during a storm is measured at ten climate stations, and have the
following values in mm:

```
45, 76, 33, 112, 67, 23, 88, 39, 71, 62
```

1. calculate the mean and the standard deviation
2. convert all observations to z-scores
3. calculate the coefficient of variation

In [2]:
rainfall_in_mm = [45, 76, 33, 112, 67, 23, 88, 39, 71, 62]

1) Let's start with the mean (μ) and the standard deviation (σ) of this list of numbers:

In [3]:
μ = sum(rainfall_in_mm) / len(rainfall_in_mm)
print(μ)

# or easier on a higher level of abstraction:
μ == np.mean(rainfall_in_mm)

61.6


True

In [4]:
σ = np.sqrt(sum([(x - μ)**2 for x in rainfall_in_mm]) / len(rainfall_in_mm))
print(σ)

# or, abstracted:
σ == np.std(rainfall_in_mm)

25.8


True

2) Next, convert the observations to z-scores:

In [5]:
z_scores_rainfall = [(x - μ) / σ for x in rainfall_in_mm]

print(z_scores_rainfall)

[-0.6434108527131783, 0.5581395348837208, -1.1085271317829457, 1.9534883720930232, 0.20930232558139528, -1.496124031007752, 1.0232558139534882, -0.875968992248062, 0.36434108527131776, 0.015503875968992192]


3) Finally, let's calculate the [coefficient of variation](https://en.wikipedia.org/wiki/Coefficient_of_variation)

In [6]:
cv_rainfall = σ / μ
print(cv_rainfall)

0.41883116883116883


The statistical measures for the rainfall at the mentioned 10 climate stations is as follows:

- **mean**: `61.6`
- **standard deviation**: `25.8`
- **coefficient of variation**: `0.41883116883116883`

The **z-scores** are as follows:

```
[-0.6434108527131783, 0.5581395348837208, -1.1085271317829457, 1.9534883720930232, 0.20930232558139528, -1.496124031007752, 1.0232558139534882, -0.875968992248062, 0.36434108527131776, 0.015503875968992192]
```

---

## Question 2

We know that the average yearly income (in Swedish crowns) in a neighborhood is normally distributed with `µ = 310 000` and `σ = 22 000`.

1. A person has a annual yearly income of 328 000. How often is it to have a salary of at least 328 000?
2. What percentage of the population in that neighborhood have an annual income lower than 270 000? 

In [39]:
salary_µ = 310_000
salary_σ = 22_000

high_income = 328_000
low_income = 270_000

1. A person has a annual yearly income of 328 000. How often is it to have a salary of at least 328 000?

In [40]:
z_score_high = (high_income - salary_μ) / salary_σ
z_score_high

0.8181818181818182

A person with an annual yearly income of 328 000 SK is (rounded) `0.82 σ` from the mean. That means, according to the [Table Under The Normal Curve](https://moodle.nateko.lu.se/pluginfile.php/14634/mod_resource/content/0/Links_and_tables/Table_area_under_normal_curve.pdf):

In [44]:
p_high = 0.2939

![Area under the curve and formulas from the course materials](area_under_curve.png)

Since we are looking for the probability for salaries _above_ the given number, which equals to the area under the curve from the value onwards, we need to substract p from 0.5:

In [49]:
p_above = 0.5 - p_high
p_above

0.2061

The probability of having an annual income of that size is `0.2061`, or approximately 20%. That means that about every 5th person has this or a higher annual income.

2. What percentage of the population in that neighborhood have an annual income lower than 270 000? 

In [41]:
z_score_low = (low_income - salary_μ) / salary_σ
z_score_low

-1.8181818181818181

The annual yearly income of 220 000 SK is (rounded) `1.82 σ` from the mean. That means, according to the [Table Under The Normal Curve](https://moodle.nateko.lu.se/pluginfile.php/14634/mod_resource/content/0/Links_and_tables/Table_area_under_normal_curve.pdf):

In [43]:
p_low = 0.4656

The probability of having an annual income of 220 000 SK or lower is `0.4656`, or approximately 46%.

---

## Question 3

The percentage of sand in 120 soil samples was measured, and the distribution of
the results is presented in this frequency table.

|Sand (%) | Frequency |
| --- | --- |
|85 | 1 |
|86 | 0 |
|87 | 8 |
|88 | 12 |
|89 | 17 |
|90 | 44 |
|91 | 19 |
|92 | 11 |
|93 | 6 |
|94 | 1 |
|95 | 1 |

a) What is the median of the percentage of sand? 

In [10]:
frequency_table = {
    85: 1,
    86: 0,
    87: 8,
    88: 12,
    89: 17,
    90: 44,
    91: 19,
    92: 11,
    93: 6,
    94: 1,
    95: 1,
}

First, we need to get a list of all the sample sand percentages to work with.

In [12]:
sample_list = []
for k, v in frequency_table.items():
    for i in range(v):
        sample_list.append(k)

In [17]:
# check whether the transform worked as expected
for k, v in frequency_table.items():
    if v != sample_list.count(k):
        raise Exception(f"Non-matching counts for {k}")
else:
    print("All good.")

All good.


The list needs to be sorted.

In [18]:
sorted_samples = sorted(sample_list)

Now we need to figure out whether we are handling an odd or even number of samples, to calculate the median accordingly by then either taking the number in the middle (odd count) or calculating the number in between the two middle ones (even count).

In [20]:
len(sorted_samples)

120

It's an even number of samples, which means we need to take the number in between the two middle ones.

In [26]:
middle_nums = sorted_samples[(int(len(sorted_samples)/2)-1):(int(len(sorted_samples)/2)+1)]  # 0-indexed
middle_nums

[90, 90]

In [30]:
median = sum(middle_nums) / len(middle_nums)

In [31]:
median

90.0

This was the complicated manual way of calculating the median, let's now do it how a sane person would do it, if they are _alread_ using `numpy`...

In [33]:
np.median(sample_list)

90.0

😁

The median of sand percentage of the given soil samples is `90.0%`.

---

## Question 4

Assuming that the height of Swedish women is normally distributed with µ = 168 cm and σ = 5.0 cm. For a special study, a research group is interested in women with a height ranging between 160-170 cm.

1) What is the percentage of all Swedish women falling within this range? 

In [45]:
height_µ = 168
height_σ = 5.0
study_height = range(160, 171)  # range() does "from x up to but not including y"

In [54]:
z_score_bottom = (study_height[0] - height_μ) / height_σ
z_score_bottom

-1.6

In [55]:
z_score_top = (study_height[-1] - height_μ) / height_σ
z_score_top

0.4

In [57]:
p_bottom = 0.4452 # lookup from z-score table

In [58]:
p_top = 0.1554  # lookup from z-score table

Now, since we want the range _between_ these two areas below the curve, we substract the higher p-score from the lower p-score:

In [62]:
p = p_bottom - p_top
p

0.28979999999999995

The percentage of Swedish women that fall into the height range of 160 - 170 cm (inclusive) and are therefore relevant for the mysterious study, lies at `28.98%`.

---

## Question 5

A distribution has a median smaller than the mean, with a large number of low observations and a small number of high observations.

1. Is this distribution positively or negatively skewed? 

![positively skewed distribution curve](positively_skewed.png)

The distribution is **positively skewed**

![](https://cdn.corporatefinanceinstitute.com/assets/Positively-Skewed-Distribution.png)